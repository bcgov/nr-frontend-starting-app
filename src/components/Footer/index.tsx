import React from 'react';
import { Container, Grid } from '@mui/material';

import FooterLink from '../FooterLink';
import FooterBox from '../FooterBox';

const Footer = () => (
  <FooterBox>
    <Container maxWidth="lg">
      <Grid
        container
        direction="row"
        justifyContent="flex-start"
        alignItems="center"
      >
        <Grid item xs="auto">
          <FooterLink data-testid="footer-home" href="#">
            Home
          </FooterLink>
        </Grid>
        <Grid item xs="auto">
          <FooterLink data-testid="footer-disclaimer" href="#">
            Disclaimer
          </FooterLink>
        </Grid>
        <Grid item xs="auto">
          <FooterLink data-testid="footer-privacy" href="#">
            Privacy
          </FooterLink>
        </Grid>
        <Grid item xs="auto">
          <FooterLink data-testid="footer-acc" href="#">
            Accessibility
          </FooterLink>
        </Grid>
        <Grid item xs="auto">
          <FooterLink data-testid="footer-copyright" href="#">
            Copyright
          </FooterLink>
        </Grid>
        <Grid item xs="auto">
          <FooterLink data-testid="footer-contact" href="#">
            Contact Us
          </FooterLink>
        </Grid>
      </Grid>
    </Container>
  </FooterBox>
);

export default Footer;
