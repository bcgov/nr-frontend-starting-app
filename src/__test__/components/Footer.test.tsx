/* eslint-disable no-undef */
import React from 'react';
import { render } from '@testing-library/react';
import renderer from 'react-test-renderer';
import Footer from '../../components/Footer';

describe('the Footer component', () => {
  it('should have the correct links', () => {
    const { getByTestId } = render(<Footer />);
    expect(getByTestId('footer-home').textContent).toBe('Home');
    expect(getByTestId('footer-disclaimer').textContent).toBe('Disclaimer');
    expect(getByTestId('footer-privacy').textContent).toBe('Privacy');
    expect(getByTestId('footer-acc').textContent).toBe('Accessibility');
    expect(getByTestId('footer-copyright').textContent).toBe('Copyright');
    expect(getByTestId('footer-contact').textContent).toBe('Contact Us');
    expect(getByTestId('footer-release').textContent).toBe('Release dev');
  });

  it('should match the snapshot', () => {
    const tree = renderer
      .create(<Footer />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
