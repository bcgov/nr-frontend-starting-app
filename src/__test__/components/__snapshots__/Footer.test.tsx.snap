// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the component renders correctly should match the snapshot 1`] = `
<footer
  className="sc-gsnTZi gyHQAh"
>
  <div
    className="MuiContainer-root MuiContainer-maxWidthLg css-1oqqzyl-MuiContainer-root"
  >
    <div
      className="MuiGrid-root MuiGrid-container css-1e2bu2o-MuiGrid-root"
    >
      <div
        className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-auto css-1wrgmsj-MuiGrid-root"
      >
        <a
          className="sc-bczRLJ keriIN"
          data-testid="footer-home"
          href="#"
        >
          Home
        </a>
      </div>
      <div
        className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-auto css-1wrgmsj-MuiGrid-root"
      >
        <a
          className="sc-bczRLJ keriIN"
          data-testid="footer-disclaimer"
          href="#"
        >
          Disclaimer
        </a>
      </div>
      <div
        className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-auto css-1wrgmsj-MuiGrid-root"
      >
        <a
          className="sc-bczRLJ keriIN"
          data-testid="footer-privacy"
          href="#"
        >
          Privacy
        </a>
      </div>
      <div
        className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-auto css-1wrgmsj-MuiGrid-root"
      >
        <a
          className="sc-bczRLJ keriIN"
          data-testid="footer-acc"
          href="#"
        >
          Accessibility
        </a>
      </div>
      <div
        className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-auto css-1wrgmsj-MuiGrid-root"
      >
        <a
          className="sc-bczRLJ keriIN"
          data-testid="footer-copyright"
          href="#"
        >
          Copyright
        </a>
      </div>
      <div
        className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-auto css-1wrgmsj-MuiGrid-root"
      >
        <a
          className="sc-bczRLJ keriIN"
          data-testid="footer-contact"
          href="#"
        >
          Contact Us
        </a>
      </div>
    </div>
  </div>
</footer>
`;
